src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
+                {
+                    // If we talk to a system we have not talked to before (or has given up talking to in the past) stop
+                    // system delivery attempts after the specified time. This act will drop the pending system messages and gate the
+                    // remote address at the EndpointManager level stopping this actor. In case the remote system becomes reachable
+                    // again it will be immediately quarantined due to out-of-sync system message buffer and becomes quarantined.
+                    // In other words, this action is safe.
+                    if (_bailoutAt != null && _bailoutAt.IsOverdue)
                     {
-                        Context.System.DeadLetters.Tell(send);
+                        throw new HopelessAssociation(_localAddress, _remoteAddress, Uid,
+                            new TimeoutException("Delivery of system messages timed out and they were dropped"));
                     }
-                })
-                .With<EndpointWriter.FlushAndStop>(flush => Context.Stop(Self))
-                .With<EndpointWriter.StopReading>(stop =>
+
+                    _writer = CreateWriter();
+                    //Resending will be triggered by the incoming GotUid message after the connection finished
+                    GoToActive();
+                }
+                else
                 {
-                    stop.ReplyTo.Tell(new EndpointWriter.StoppedReading(stop.Writer));
-                    Sender.Tell(new EndpointWriter.StoppedReading(stop.Writer));
-                });
+                    GoToIdle();
+                }
+            });
+            Receive<AttemptSysMsgRedelivery>(redelivery => { }); // Ignore
+            Receive<EndpointManager.Send>(send => send.Message is ISystemMessage, send => TryBuffer(send.Copy(NextSeq())));
+            Receive<EndpointManager.Send>(send => Context.System.DeadLetters.Tell(send));
+            Receive<EndpointWriter.FlushAndStop>(flush => Context.Stop(Self));
+            Receive<EndpointWriter.StopReading>(stop =>
+            {
+                stop.ReplyTo.Tell(new EndpointWriter.StoppedReading(stop.Writer));
+                Sender.Tell(new EndpointWriter.StoppedReading(stop.Writer));
+            });
         }
 
-        protected void Idle(object message)
+        protected void IdleBehavior()
         {
-            message.Match()
-                .With<EndpointManager.Send>(send =>
-                {
-                    _writer = CreateWriter();
-                    //Resending will be triggered by the incoming GotUid message after the connection finished
-                    HandleSend(send);
-                    Context.Become(OnReceive);
-                })
-                .With<AttemptSysMsgRedelivery>(sys =>
+            Receive<IsIdle>(idle => Sender.Tell(Idle.Instance));
+            Receive<EndpointManager.Send>(send =>
```

Comment:
If we receive a `Send` in the idle state, this will trigger the `ReliableDeliverySupervisor` to move back into an active state.